lib/std/time.zig

```diff
@@ -59,6 +59,395 @@ pub fn sleep(nanoseconds: u64) void {
     std.os.nanosleep(s, ns);
 }
 
+pub const TimeZone = @import("time/TimeZone.zig");
+
+pub const LocalTimeError = error{
+    /// The local time zone of the system could not be determined.
+    LocalTimeUnavailable,
+} || std.mem.Allocator.Error;
+
+pub fn localtime(allocator: std.mem.Allocator) LocalTimeError!TimeZone {
+    switch (builtin.os.tag) {
+        .linux => {
+            var tz_file = std.fs.openFileAbsolute("/etc/localtime", .{}) catch
+                return error.LocalTimeUnavailable;
+            defer tz_file.close();
+
+            var buffered = std.io.bufferedReader(tz_file.reader());
+            return TimeZone.parse(allocator, buffered.reader()) catch
+                return error.LocalTimeUnavailable;
+        },
+        else => @compileError("`localtime` has not been implemented for this OS!"),
+    }
+}
+
+/// ISO 8601 compliant date and time representation.
+pub const DateTime = struct {
+    /// nanoseconds, range 0 to 999
+    nanosecond: u16,
+
+    /// microseconds, range 0 to 999
+    microsecond: u16,
+
+    /// milliseconds, range 0 to 999
+    millisecond: u16,
+
+    /// seconds, range 0 to 60
+    second: u8,
+
+    /// minutes, range 0 to 59
+    minute: u8,
+
+    /// hours, range 0 to 23
+    hour: u8,
+
+    /// day of the month, range 1 to 31
+    day: u8,
+
+    /// day of the year, range 1 to 366
+    year_day: u16,
```

Comment:
regarding year_day and week_day - what's the reason to carry those around with any datetime instance? why not calculate them "on demand"?

Reply:
Why carry anything but the timestamp and the timezone offset? Either way removing them doesn't affect the size of the struct.

Reply:
I think both Go takes the timestamp approach. Rust's Chrono seems to be in the middle. I feel that one benefit of a timestamp approach is that it's easier to implement date arithmetic.

Reply:
I didn't mean to advocate a purely timestamp-based approach. If there is a need to have fields like `year_day` stored in the structure then that's fine. I was just thinking that fields that are rarely required could be replaced by methods that calculate the value on request for given datetime.